pkg/trino/driver/driver.go

```diff
@@ -35,11 +37,43 @@ func Open(settings models.TrinoDatasourceSettings) (*sql.DB, error) {
 			sslCertPath = f.Name()
 			defer os.Remove(f.Name())
 		}
+
+		if settings.Opts.TLS.ClientCertificate != "" {
+			if settings.Opts.TLS.ClientKey == "" {
+				return nil, errors.New("client certificate was configured without a client key")
+			}
+			clientCertF, certErr := os.CreateTemp("", "client-cert")
```

Comment:
To get this plugin certified by Grafana, we can't use any IO here, so we can't create temporary files. In the latest `trino-go-client` version, we allowed passing the SSL cert as a string. See if we also need to make some changes to pass client certs like this.

Reply:
Or can you use `X509KeyPair` instead of `LoadX509KeyPair`?

Reply:
thanks Jan, I have used `X509KeyPair` instead. I like that a custom http client can be provided to the trino go client as that allows for nice features like certificate reloading using `GetCertificate`.